.github/workflows/ubuntu.yml

```diff
@@ -37,23 +36,16 @@ jobs:
         run: |
           sudo apt-get update
           sudo apt-get -y install zlib1g-dev libgmp3-dev
-          sudo apt-get -y install ${{ matrix.compiler-version }}
-          export CC=${{ matrix.compiler-version }}
-          if [[ $CC == gcc* ]]; then
-            export CXX="$(echo "${CC}" | sed "s/cc/++/g")";
-          elif [[ $CC == clang* ]]; then
-            export CXX="$(echo "${CC}" | sed "s/clang/clang++/g")";
-          else
-            echo "Unknown compiler version";
-            exit 1;
-          fi
+          sudo apt-get -y install ${{ matrix.compiler-version.cc }}
           mkdir /home/runner/work/downward/lib
 
       # We only want to setup osi if both LP solvers are set, hence
       # we execute the following three steps only if both secrets
       # are set.
       - name: setup-cplex
         env:
+          CC: ${{ matrix.compiler-version.cc }}
+          CXX: ${{ matrix.compiler-version.cxx }}
```

Comment:
I think you could also set an environment variable that applies to all steps in a single place: https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions#setting-an-environment-variable

Reply:
As Jendrik said, you can do this jsut at the top of the script. You can take a look at my issue1005